File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        first = 0
        sec = len(numbers)-1
        
        while first < sec:
            if numbers[first] + numbers[sec] == target:
                return [first+1, sec+1]
            elif numbers[first] + numbers[sec] < target:
                first += 1
            elif numbers[first] + numbers[sec] > target:
                sec -= 1

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        
        int r = numbers.length - 1;
        
        for(int l=0; l<numbers.length; l++)
        {
            int need = target - numbers[l];
            
            while(numbers[r] > need)
                r--;
            
            if(l < r && numbers[r] == need)
            {
                int[] answer = {l +1, r+1};
                return answer;
            }
                 
        }
        
        return null;
        
    }
}